src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,170 @@
+    }
+
+    /**
+     * Outputs arm subsystem
+     *
+     */
+    override fun outputToSmartDashboard() {
+        SmartDashboard.putNumber("armPower", armPower)
+    }
+
+    @Synchronized override fun stop() {
+        movementState = MovementState.STATIONARY
+        setArmVelocity(0.0)
+        brake.set(DoubleSolenoid.Value.kReverse)
+    }
+
+    /**
+     * Handles power to arm during each loop cycle
+     *
+     * @constructor Creates loop that controls power to arm during each loop cycle
+     */
+    val loop: Loop = object : Loop {
+
+        override fun onStart() {
+            movementState = MovementState.STATIONARY
+            setArmVelocity(0.0)
+            brake.set(DoubleSolenoid.Value.kForward)
+        }
+
+        /**
+         * Handles events during each loop cycle, depends on arm control mode
+         */
+        override fun onLoop() {
+            synchronized(this@Arm) {
+                when (armState) {
+                    Arm.ArmState.LOW -> {
+                        setArmPosition(ArmState.LOW.targetPos)
+                    }
+                    Arm.ArmState.EXCHANGE -> {
+                        setArmPosition(ArmState.EXCHANGE.targetPos)
+                    }
+                    Arm.ArmState.HIGH -> {
+                        setArmPosition(ArmState.HIGH.targetPos)
+                    }
+                    //masterSRX.sensorCollection.quadratureVelocity
+                    Arm.ArmState.VELOCITY_CONTROL -> { }
+                }
+                if ((armState == Arm.ArmState.LOW && masterSRX.sensorCollection.quadratureVelocity > 0) || (armState == Arm.ArmState.HIGH && masterSRX.sensorCollection.quadratureVelocity < 0)) {
+                    movementState = Arm.MovementState.MOVING_TO_EXCHANGE
+                } else if (armState == Arm.ArmState.EXCHANGE && masterSRX.sensorCollection.quadratureVelocity > 0) {
+                    movementState = Arm.MovementState.MOVING_TO_TOP
+                } else if (armState == Arm.ArmState.EXCHANGE && masterSRX.sensorCollection.quadratureVelocity < 0) {
+                    movementState = Arm.MovementState.MOVING_TO_BOTTOM
+                } else if (masterSRX.sensorCollection.quadratureVelocity > 0) {
+                    movementState = Arm.MovementState.UP
+                } else if (masterSRX.sensorCollection.quadratureVelocity == 0) {
+                    stationaryTime ++
+                    if (stationaryTime >= 100) {
+                        movementState = Arm.MovementState.STATIONARY
+                    }
+                } else {
+                    movementState = Arm.MovementState.DOWN
+                }
+                armAngle = ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition) - armBaseAngle
+                if (movementState == Arm.MovementState.STATIONARY) {
+                    brake.set(DoubleSolenoid.Value.kReverse)
```

Comment:
The brake is actuated in stationary, but is never actually un-actuated when the arm starts moving. Additionally, it might be a good idea to add a timer that prevents the arm from moving for some amount of time after the brake disengages - although that may be something to implement at a later date.